### PR TITLE
Updated Testcases for RemoteModel.java

### LINKED ISSUE #1382: [FEATURE]Improve test coverage for RemoteModel.java
We need to improve test coverage in our repo.

This issue will target only to improve test coverage for RemoteModel.java

From this url we can see where test coverage is missing for this class: https://app.codecov.io/gh/opensearch-project/ml-commons/blob/2.x/ml-algorithms%2Fsrc%2Fmain%2Fjava%2Forg%2Fopensearch%2Fml%2Fengine%2Falgorithms%2Fremote%2FRemoteModel.java

### PR DESCRIPTION
**Pull request by divitr**
### Description
Updated Testcases for RemoteModel.java
 
### Issues Resolved
Resolves #1382
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by austintlee** [comment]
@dhrubo-os do you know why coverage went _down_ with an addition of a test case?

File: ml-algorithms/src/test/java/org/opensearch/ml/engine/algorithms/remote/RemoteModelTest.java [hunk 1]
```diff
@@ -65,6 +73,15 @@ public void predict_NullConnectorExecutor() {
         remoteModel.predict(mlInput);
     }
 
+    @Test
+    public void predict_ModelPredictSuccess() {
+        Connector connector = createConnector(ImmutableMap.of("Authorization", "Bearer ${credential.key}"));
+        when(mlModel.getConnector()).thenReturn(connector);
+        remoteModel.initModel(mlModel, ImmutableMap.of(), encryptor);
+        Assert.assertTrue(remoteModel.isModelReady());
```
Comment (austintlee) [inline ml-algorithms/src/test/java/org/opensearch/ml/engine/algorithms/remote/RemoteModelTest.java:81]: Is this going to be flaky due to timing?

File: ml-algorithms/src/test/java/org/opensearch/ml/engine/algorithms/remote/RemoteModelTest.java [hunk 2]
```diff
@@ -65,6 +73,15 @@ public void predict_NullConnectorExecutor() {
         remoteModel.predict(mlInput);
     }
 
+    @Test
+    public void predict_ModelPredictSuccess() {
```
Comment (austintlee) [inline ml-algorithms/src/test/java/org/opensearch/ml/engine/algorithms/remote/RemoteModelTest.java:77]: "testXYZ" and no underscore.

File: ml-algorithms/src/test/java/org/opensearch/ml/engine/algorithms/remote/RemoteModelTest.java [hunk 3]
```diff
@@ -65,6 +73,15 @@ public void predict_NullConnectorExecutor() {
...
+        Connector connector = createConnector(ImmutableMap.of("Authorization", "Bearer ${credential.key}"));
+        when(mlModel.getConnector()).thenReturn(connector);
+        remoteModel.initModel(mlModel, ImmutableMap.of(), encryptor);
+        Assert.assertTrue(remoteModel.isModelReady());
+        when(remoteConnectorExecutor.executePredict(mlInput)).thenReturn(new ModelTensorOutput(new ArrayList<>()));
```
Comment (austintlee) [inline ml-algorithms/src/test/java/org/opensearch/ml/engine/algorithms/remote/RemoteModelTest.java:82]: What are you testing??

**Review by austintlee — CHANGES_REQUESTED** [review]
I don't think you are testing the right thing here.

**Review by Zhangxunmt — COMMENTED** [review]
I am going to close this PR if there's still no response to Austin's comments. @divitr

**Pull request closed without merging**